pkg/service/service.go

```diff
@@ -176,7 +178,8 @@ func (svc *svcInfo) filterBackends(frontend lb.L3n4AddrID) bool {
 }
 
 func (svc *svcInfo) useMaglev() bool {
-	if option.Config.NodePortAlg != option.NodePortAlgMaglev {
+	if option.Config.NodePortAlg != option.NodePortAlgMaglev &&
```

Comment:
Shouldn't this check `svc.LoadBalancerAlgo`? Maintaining the maglev table is very expensive and it should only be done if requested.

Reply:
good catch! fixed
I removed checking for option.Config because svc.loadBalancerAlgo is always populated (if annotation is not supported the value is copied from options)